### PR TITLE
fix #78 - Add processor_usage_reporting_interval_seconds and file_poll_interval_seconds config params

### LINKED ISSUE #78: Add processor_usage_reporting_interval_seconds and file_poll_interval_seconds config params
Extend the `LogSearchShipper` config element to enable setting:

 * `processor_usage_reporting_interval_seconds` - affects how often the [\*ProcessorUsage\* metrics are reported](https://github.com/cityindex/LogSearchShipper/blob/master/src/LogsearchShipper.Core/NxLog/NxLogProcessManager.cs#L178)
 * `file_poll_interval_seconds` - affects [nxlog's `PollInterval` config setting's](https://github.com/cityindex/LogSearchShipper/blob/master/src/LogsearchShipper.Core/NxLog/NxLogProcessManager.cs#L601) value.  nxlog's `DirCheckInterval` setting should be set to 2 * `file_poll_interval_seconds`

eg:

```xml
 <LogSearchShipper data_folder="data" ingestor_host="ingestor.you-logsearch-cluster.com" 
ingestor_port="443" shipper_service_username="" shipper_service_password="" 
processor_usage_reporting_interval_seconds="1" file_poll_interval_seconds="5"
>
```

### PR DESCRIPTION
**Pull request by fandrei**
Addresses #78

### CODE REVIEW

**Comment by le-bott** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.meta.logsearch.io/job/shared.releases.logsearchshipper-PRs/58/

**Build Log**
*last 50 lines*

 ```
[...truncated 143 lines...]
  Deleting file "C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\IntegrationTests\obj\Release\IntegrationTests.pdb".
RestorePackages:
  "C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\.nuget\NuGet.exe" install "C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\IntegrationTests\packages.config" -source ""  -NonInteractive  -solutionDir "C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\ " 
  All packages listed in packages.config are already installed.
Done Building Project "C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\IntegrationTests\IntegrationTests.csproj" (Rebuild target(s)) -- FAILED.
Project "C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\LogSearchShipper.sln" (2) is building "C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\IntegrationTests\DummyServiceWithLogRolling\DummyServiceWithLogRolling.csproj" (8) on node 1 (Rebuild target(s)).
Project file contains ToolsVersion="12.0". This toolset may be unknown or missing, in which case you may be able to resolve this by installing the appropriate version of MSBuild, or the build may have been forced to a particular ToolsVersion for policy reasons. Treating the project as if it had ToolsVersion="4.0". For more information, please see http://go.microsoft.com/fwlink/?LinkId=291333.
CoreClean:
  Deleting file "C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\IntegrationTests\bin\DummyServiceWithLogRolling.exe.config".
  Deleting file "C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\IntegrationTests\bin\DummyServiceWithLogRolling.exe".
  Deleting file "C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\IntegrationTests\bin\DummyServiceWithLogRolling.pdb".
  Deleting file "C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\IntegrationTests\DummyServiceWithLogRolling\obj\Release\DummyServiceWithLogRolling.exe".
  Deleting file "C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\IntegrationTests\DummyServiceWithLogRolling\obj\Release\DummyServiceWithLogRolling.pdb".
RestorePackages:
  "C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\.nuget\NuGet.exe" install "C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\IntegrationTests\DummyServiceWithLogRolling\packages.config" -source ""  -NonInteractive  -solutionDir "C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\ " 
  All packages listed in packages.config are already installed.
GenerateTargetFrameworkMonikerAttribute:
Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output files are up-to-date with respect to the input files.
CoreCompile:
  C:\Windows\Microsoft.NET\Framework\v4.0.30319\Csc.exe /noconfig /nowarn:1701,1702 /nostdlib+ /platform:anycpu32bitpreferred /errorreport:prompt /warn:4 /define:TRACE /highentropyva+ /reference:C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\packages\log4net.2.0.3\lib\net40-full\log4net.dll /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\Microsoft.CSharp.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\mscorlib.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\System.Core.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\System.Data.DataSetExtensions.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\System.Data.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\System.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\System.Xml.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\System.Xml.Linq.dll" /reference:C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\packages\Topshelf.3.1.3\lib\net40-full\Topshelf.dll /reference:C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\packages\Topshelf.Log4Net.3.1.3\lib\net40-full\Topshelf.Log4Net.dll /debug:pdbonly /filealign:512 /optimize+ /out:obj\Release\DummyServiceWithLogRolling.exe /subsystemversion:6.00 /target:exe /utf8output Program.cs Properties\AssemblyInfo.cs "C:\Windows\TEMP\.NETFramework,Version=v4.5.AssemblyAttributes.cs"
_CopyFilesMarkedCopyLocal:
  Copying file from "C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\packages\log4net.2.0.3\lib\net40-full\log4net.dll" to "..\bin\log4net.dll".
  Copying file from "C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\packages\Topshelf.3.1.3\lib\net40-full\Topshelf.dll" to "..\bin\Topshelf.dll".
  Copying file from "C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\packages\Topshelf.Log4Net.3.1.3\lib\net40-full\Topshelf.Log4Net.dll" to "..\bin\Topshelf.Log4Net.dll".
  Copying file from "C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\packages\log4net.2.0.3\lib\net40-full\log4net.xml" to "..\bin\log4net.xml".
_CopyAppConfigFile:
  Copying file from "App.config" to "..\bin\DummyServiceWithLogRolling.exe.config".
CopyFilesToOutputDirectory:
  Copying file from "obj\Release\DummyServiceWithLogRolling.exe" to "..\bin\DummyServiceWithLogRolling.exe".
  DummyServiceWithLogRolling -> C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\IntegrationTests\bin\DummyServiceWithLogRolling.exe
  Copying file from "obj\Release\DummyServiceWithLogRolling.pdb" to "..\bin\DummyServiceWithLogRolling.pdb".
Done Building Project "C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\IntegrationTests\DummyServiceWithLogRolling\DummyServiceWithLogRolling.csproj" (Rebuild target(s)).
Done Building Project "C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\LogSearchShipper.sln" (Rebuild target(s)) -- FAILED.
Done Building Project "C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\build.xml.proj" (Build;RunTests;Package target(s)) -- FAILED.

Build FAILED.

"C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\build.xml.proj" (Build;RunTests;Package target) (1) ->
"C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\LogSearchShipper.sln" (Rebuild target) (2) ->
"C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\LogsearchShipper.Core\LogSearchShipper.Core.csproj" (Rebuild target) (3) ->
(CoreCompile target) -> 
  LogsearchShipperProcessManager.cs(46,70): error CS1061: 'LogSearchShipper.Core.ConfigurationSections.LogSearchShipperSection' does not contain a definition for 'ProcessorUsageReportingIntervalSeconds' and no extension method 'ProcessorUsageReportingIntervalSeconds' accepting a first argument of type 'LogSearchShipper.Core.ConfigurationSections.LogSearchShipperSection' could be found (are you missing a using directive or an assembly reference?) [C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\LogsearchShipper.Core\LogSearchShipper.Core.csproj]
  LogsearchShipperProcessManager.cs(47,55): error CS1061: 'LogSearchShipper.Core.ConfigurationSections.LogSearchShipperSection' does not contain a definition for 'FilePollIntervalSeconds' and no extension method 'FilePollIntervalSeconds' accepting a first argument of type 'LogSearchShipper.Core.ConfigurationSections.LogSearchShipperSection' could be found (are you missing a using directive or an assembly reference?) [C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\src\LogsearchShipper.Core\LogSearchShipper.Core.csproj]

    0 Warning(s)
    2 Error(s)

Time Elapsed 00:00:04.32
Build step 'Build a Visual Studio project or solution using MSBuild' marked build as failure
```
Test FAILed.

**Comment by le-bott** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.meta.logsearch.io/job/shared.releases.logsearchshipper-PRs/59/

**Build Log**
*last 50 lines*

 ```
[...truncated 211 lines...]
  Options:
  /fixture=STR         	Test fixture or namespace to be loaded (Deprecated) (Short format: /load=STR)
  /run=STR             	Name of the test case(s), fixture(s) or namespace(s) to run
  /runlist=STR         	Name of a file containing a list of the tests to run, one per line
  /config=STR          	Project configuration (e.g.: Debug) to load
  /result=STR          	Name of XML result file (Default: TestResult.xml) (Short format: /xml=STR)
  /xmlConsole          	Display XML to the console (Deprecated)
  /noresult            	Suppress XML result output (Short format: /noxml)
  /output=STR          	File to receive test output (Short format: /out=STR)
  /err=STR             	File to receive test error output
  /work=STR            	Work directory for output files
  /labels              	Label each test in stdOut
  /trace=X             	Set internal trace level: Off, Error, Warning, Info, Verbose
  /include=STR         	List of categories to include
  /exclude=STR         	List of categories to exclude
  /framework=STR       	Framework version to be used for tests
  /process=X           	Process model for tests: Single, Separate, Multiple
  /domain=X            	AppDomain Usage for tests: None, Single, Multiple
  /apartment=X         	Apartment for running tests: MTA (Default), STA
  /noshadow            	Disable shadow copy when running in separate domain
  /nothread            	Disable use of a separate thread for tests
  /basepath=STR        	Base path to be used when loading the assemblies
  /privatebinpath=STR  	Additional directories to be probed when loading assemblies, separated by semicolons
  /timeout=X           	Set timeout for each test case in milliseconds
  /wait                	Wait for input before closing console window
  /nologo              	Do not display the logo
  /nodots              	Do not display progress
  /stoponerror         	Stop after the first test failure or error
  /cleanup             	Erase any leftover cache files and exit
  /help                	Display help (Short format: /?)
  
  
  Options that take values may use an equal sign, a colon
  or a space to separate the option from its value.
  
C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\build.xml.proj(49,3): error MSB3073: The command ""tools\NUnit.Runners.lite.2.6.3.20131019\nunit-console.exe"  -result=TestResults.xml -out=TestOutput.log" exited with code -1.
Done Building Project "C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\build.xml.proj" (Build;RunTests;Package target(s)) -- FAILED.

Build FAILED.

"C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\build.xml.proj" (Build;RunTests;Package target) (1) ->
(RunTests target) -> 
  C:\jenkins\workspace\shared.releases.logsearchshipper-PRs\build.xml.proj(49,3): error MSB3073: The command ""tools\NUnit.Runners.lite.2.6.3.20131019\nunit-console.exe"  -result=TestResults.xml -out=TestOutput.log" exited with code -1.

    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:05.75
Build step 'Build a Visual Studio project or solution using MSBuild' marked build as failure
```
Test FAILed.

**Comment by le-bott** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.meta.logsearch.io/job/shared.releases.logsearchshipper-PRs/60/Test PASSed.

**Comment by fandrei** [comment]
@mrdavidlaing, what do you think about this?

File: src/LogsearchShipper.Core/NxLog/NxLogProcessManager.cs [hunk 1]
```diff
@@ -598,14 +602,15 @@ InputType	multiline
 	ReadFromLast {2}
 	SavePos	TRUE
 	CloseWhenIdle TRUE
-	PollInterval 5
+	PollInterval {5}
 	DirCheckInterval 10
```
Comment (mrdavidlaing) [inline src/LogsearchShipper.Core/NxLog/NxLogProcessManager.cs]: @fandrei - could you make the `DirCheckInterval` = `2 * PollInterval` please

Comment (mrdavidlaing) [inline src/LogsearchShipper.Core/NxLog/NxLogProcessManager.cs]: And add a test to verify that the when you ser in `NxLogProcessManager.FilePollIntervalSeconds` to a `x` then the nxlog config has `PollInterval {x}` and `DirCheckInterval {2x}`

**Comment by le-bott** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.meta.logsearch.io/job/shared.releases.logsearchshipper-PRs/61/Test PASSed.

**Comment by fandrei** [comment]
@mrdavidlaing, done

**Comment by le-bott** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.meta.logsearch.io/job/shared.releases.logsearchshipper-PRs/62/Test PASSed.

**Comment by fandrei** [comment]
@mrdavidlaing, added a test

File: src/LogsearchShipper.Core/ConfigurationSections/LogsearchShipperSection.cs [hunk 2]
```diff
@@ -58,5 +58,19 @@ public string SessionId
 			get { return (string)this["sessionId"]; }
 			set { this["sessionId"] = value; }
 		}
+
+		[ConfigurationProperty("processor_usage_reporting_interval_seconds", IsRequired = false, DefaultValue = 1.0d)]
```
Comment (mrdavidlaing) [inline src/LogsearchShipper.Core/ConfigurationSections/LogsearchShipperSection.cs]: @fandrei - I think this default should remain at the current 60

**Comment by mrdavidlaing** [comment]
@fandrei - after changing the default; lets get this merged so I can deploy it later today

**Comment by le-bott** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.meta.logsearch.io/job/shared.releases.logsearchshipper-PRs/63/Test PASSed.

**Comment by fandrei** [comment]
@mrdavidlaing, done

**Comment by fandrei** [comment]
@mrdavidlaing, should I merge it?

**Comment by mrdavidlaing** [comment]
@fandrei - done!

**Pull request merged**